### ISSUE TITLE
Add file to playlist: "Unable to find linked item at path ..." in log - I believe it's due to apostrophe in name of file

### ISSUE THREAD

**Issue by smathev** [jellyfin/jellyfin#3936]
**Describe the bug**
Adding file with apostrophe in filename to m3u8 / m3u playlist results in: 
"
Unable to find linked item at path file:///storage/data/music/artists/Coheed.and.Cambria/Good.Apollo.I’m.Burning.Star.IV.Volume.One.From.Fear.Through.the.Eyes.of.Madness-(2005)/CD.02/01 - Welcome Home.flac/storage/data/music/artists/Coheed.and.Cambria/Good.Apollo.I’m.Burning.Star.IV.Volume.One.From.Fear.Through.the.Eyes.of.Madness-(2005)/CD.02/01 - Welcome Home.flac
"
in the log.

**To Reproduce**
1. Have one or more files with an apostrophe in the filename.
2. Find file in relevant library
3. Try to add file to playlist
4. See error in logs.

**Expected behavior**
Jellyfin should be able to handle single quotes/apostrophes in filenames.

**System info (please complete the following information):**
 - OS: Ubuntu
 - Virtualization: Docker
 - Clients: Browser
 - Browser: Chrome  84.0.4147.125
 - Jellyfin Version: 10.6.3
 - Playback: N/A
 - Installed Plugins: 
 - Reverse Proxy: nginx
 - Base URL: N/A
 - Networking: Host
 - Storage: mergerFS

**Additional context**
I'm fairly certain it's due to there being apostrophe in the filename. That apostrophe makes the file-location become a duplication of itself. 
I'm able to add all other files I have in my libraries, but all with an apostrophe fails in the same way:

Unable to find linked item at path file:///storage/data/music/artists/Beastie.Boys/Hello.Nasty-(1998)/06-Body.Movin’.flac/storage/data/music/artists/Beastie.Boys/Hello.Nasty-(1998)/06-Body.Movin’.flac
Unable to find linked item at path file:///storage/data/music/artists/Beastie.Boys/Hello.Nasty-(1998)/08-Sneakin’.Out.the.Hospital.flac/storage/data/music/artists/Beastie.Boys/Hello.Nasty-(1998)/08-Sneakin’.Out.the.Hospital.flac
Unable to find linked item at path file:///storage/data/music/artists/Red.Hot.Chili.Peppers/By.the.Way-(2001)/05-Don’t.Forget.Me.flac/storage/data/music/artists/Red.Hot.Chili.Peppers/By.the.Way-(2001)/05-Don’t.Forget.Me.flac
Unable to find linked item at path file:///storage/data/music/artists/Beastie.Boys/Hello.Nasty-(1998)/10-Flowin’.Prose.flac/storage/data/music/artists/Beastie.Boys/Hello.Nasty-(1998)/10-Flowin’.Prose.flac
Unable to find linked item at path file:///storage/data/music/artists/Red.Hot.Chili.Peppers/By.the.Way-(2001)/07-Can’t.Stop.flac/storage/data/music/artists/Red.Hot.Chili.Peppers/By.the.Way-(2001)/07-Can’t.Stop.flac

For reference, the files play without issue when I start playing them.

**Comment by ferferga** [comment]
Can you confirm this is the same as #1907?

You can add more details if it's similar but not exactly the same so we can close duplicates.

**Comment by smathev** [comment]
@ferferga By my best guesses the two issues arent related.

The linked issue is related to moved files, as far as I can tell.

My issues relates to filenames having an apostrophe in the filename.

My examples are filenames in an album folder. I can add all files without apostrophe/single quote to a playlist. But all files with apostrophe/single quote will fail.

**Comment by cvium** [comment]
@smathev I was able to replicate it... somewhat. I have a file with `’` in its filename and a reference to that file in an m3u playlist, _but_ the filename in the playlist is actually `'`. The former is a "Typesetter's apostrophe" (U+2019) and the latter is "Typewriter apostrophe" (U+0027).

The two apostrophes are not the same character code and thus it doesn't work. Can you verify that the characters are identical?

**Comment by smathev** [comment]
Hi.

I'm guessing that's the issue. I have a hard time actually knowing for
sure, as I'm unsure how I should replicate it. I've tried renaming the
files in question, replacing the apostrophe with a new apostrophe from my
keyboard, but the issue persists. I'm using a danish layout, and using the
´ apostrophe still gives the error.

For reference, I'm using an ErgoTravel, on the QMK-firmware, sending the
NO_ACUT signal. I have a hard time determining which apostrophe that is
sending, but I found some documentation on reddit from the author of some
of the firmware:
https://www.reddit.com/r/olkb/comments/44804d/backslash_on_the_swedishnordic_layout/

I hope that helps.


Best regards / Med venlig hilsen

*Sigurd Mathiasen*
